src/contract/client.ts

```diff
@@ -20,15 +58,74 @@ import { processSpecEntryStream } from './utils';
  * @param {ClientOptions} options see {@link ClientOptions}
  */
 export class Client {
+  static async deploy<T = Client>(
+    args: Record<string, any> | null,
+    options: MethodOptions &
+      Omit<ClientOptions, "contractId"> & {
+        /** The hash of the wasm blob, which must already be deployed on-chain. */
+        wasmHash: Buffer | string;
+        salt?: Buffer;
+        format?: "hex" | "base64";
+      }
+  ): Promise<AssembledTransaction<T>> {
+    const spec = await specFromWasmHash(
+      options.wasmHash,
+      options,
+      options.format
+    );
+    const wasmHashBuffer =
+      typeof options.wasmHash === "string"
+        ? Buffer.from(options.wasmHash, options.format ?? "hex")
+        : (options.wasmHash as Buffer);
+    const constructorArgs: xdr.ScVal[] = args
+      ? spec.funcArgsToScVals(CONSTRUCTOR_FUNC, args)
+      : [];
+    const address = new Address(options.publicKey!);
+    const contractIdPreimage =
+      xdr.ContractIdPreimage.contractIdPreimageFromAddress(
+        new xdr.ContractIdPreimageFromAddress({
+          salt: options.salt ?? hash(randomBytes(48)),
+          address: address.toScAddress(),
+        })
+      );
+    const contractId = Address.contract(
+      hash(contractIdPreimage.toXDR())
+    ).toString();
+    const func = xdr.HostFunction.hostFunctionTypeCreateContractV2(
+      new xdr.CreateContractArgsV2({
+        constructorArgs,
+        contractIdPreimage,
+        executable:
+          xdr.ContractExecutable.contractExecutableWasm(wasmHashBuffer),
+      })
+    );
+    const operation = Operation.invokeHostFunction({
+      func,
+      source: address.toString(),
+    });
```

Comment:
Just another reminder that I'm pretty sure this is exactly what is already supported by `Operation.createCustomContract` (see https://github.com/stellar/js-stellar-base/pull/770): 

```typescript
const operation = Operation.createCustomContract({
  address: new Address(options.publicKey!),
  wasmHash: wasmHashBuffer,
  salt: options.salt,
  constructorArgs
});
```